aws-sns-topicpolicy/src/main/java/software/amazon/sns/topicpolicy/CreateHandler.java

```diff
@@ -1,77 +1,74 @@
 
-    @Override
     protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final AmazonWebServicesClientProxy proxy,
             final ResourceHandlerRequest<ResourceModel> request,
             final CallbackContext callbackContext,
             final ProxyClient<SnsClient> proxyClient,
             final Logger logger) {
 
-        final ResourceModel model = request.getDesiredResourceState();
-        // Check if invalid request
-        if (CollectionUtils.isNullOrEmpty(model.getPolicyDocument())
-                || CollectionUtils.isNullOrEmpty(model.getTopics()))
-        {
-            throw new CfnInvalidRequestException(
-                    String.format("Invalid create request, topics & policy document cannot be null or empty : %s)",
-                            model.toString()));
+        this.logger = logger;
+        ResourceModel resourceModel = request.getDesiredResourceState();
+
+        if (resourceModel == null) {
+            return ProgressEvent.failed(resourceModel, callbackContext, null, "Property PolicyDocument cannot be empty");
+        }
+        else if (CollectionUtils.isNullOrEmpty(resourceModel.getTopics())) {
+            return ProgressEvent.failed(resourceModel, callbackContext, null, "Value of property Topics must be of type List of String");
+        }
+        else if (CollectionUtils.isNullOrEmpty(resourceModel.getPolicyDocument())) {
+            return ProgressEvent.failed(resourceModel, callbackContext, HandlerErrorCode.InvalidRequest, "Invalid parameter: Policy Error: null");
         }
-        return ProgressEvent.progress(model, callbackContext)
-                .then(progress -> initCallbackContextAndPrimaryIdentifier(proxy, proxyClient, request, callbackContext,
-                        progress))
-                .then(progress -> doCreate(proxy, proxyClient, request, progress, ACTION_CREATED, CREATE_HANDLER, logger ))
-                .then(progress -> ProgressEvent.success(model, callbackContext));
-    }
 
-    /**
-     * Invocation of initCallbackContextAndPrimaryIdentifier generates primary identifier, initializes the context &
-     * stores topics in it.
-     *
-     * @param request
-     *            {@link ResourceHandlerRequest}
-     * @param callbackContext
-     *            {@link CallbackContext}
-     * @return {@link ProgressEvent}
-     */
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling Create SNS TopicPolicy", request.getStackId(), request.getClientRequestToken()));
+
+        return ProgressEvent.progress(resourceModel, callbackContext)
+                .then(progress -> Create(proxy, proxyClient, request, progress, logger))
+                .then(progress -> ProgressEvent.defaultSuccessHandler(null));
+    }
 
-    private ProgressEvent<ResourceModel, CallbackContext> initCallbackContextAndPrimaryIdentifier(
+    protected ProgressEvent<ResourceModel, CallbackContext> Create(
```

Comment:
There is no contract test for Topicpolicy. Is existence check required?

Reply:
https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/contract-tests.html 
Above link is the public doc about the contract tests.

Internally we are using Contract Test V2 already, it is sharing the same concept, but different procedure of executing and more test cases. Are you aware of this? Maybe you should publish the code change in internal system then I can leave with more detailed here. Since this is not yet migrated, we can have all the detailed in internal GitFarm first, merged there and move code to GH.

Reply:
Yes, this resource has an exception for ctv2. We don't need to do contract tests.